src/Control/Distributed/Process/Internal/Messaging.hs

```diff
@@ -135,24 +137,28 @@ connBetween node from to implicitReconnect = do
         setupConnBetween node from to implicitReconnect
 
 disconnect :: LocalNode -> Identifier -> Identifier -> IO ()
-disconnect node from to =
-  modifyValidLocalState_ node $ \vst ->
+disconnect node from to = do
```

Comment:
We want the connection to be closed eventually, but masking assumes that NT.close is uninterruptible or it won't accomplish anything. And I don't think NT.close is uninterruptible.

Maybe mask and close the connection with `forkIO $ forM_ mio id`?